### PR TITLE
Fix some clippy pedantic findings

### PR DESCRIPTION
**Pull request by Dr-Emann**
See individual commits for the specific clippy findings fixed

### CODE REVIEW

File: src/commands/prune.rs [hunk 1]
```diff
@@ -589,9 +589,9 @@ impl Pruner {
         };
 
         self.repack_candidates.sort_unstable_by_key(|rc| rc.0);
-        let mut resize_packs: BlobTypeMap<Vec<_>> = Default::default();
-        let mut do_repack: BlobTypeMap<bool> = Default::default();
-        let mut repack_size: BlobTypeMap<u64> = Default::default();
+        let mut resize_packs: BlobTypeMap<Vec<_>> = BlobTypeMap::default();
+        let mut do_repack: BlobTypeMap<bool> = BlobTypeMap::default();
```
Comment (aawsome) [inline src/commands/prune.rs:593]: Can't we remove the type hint here?

File: src/commands/prune.rs [hunk 2]
```diff
@@ -589,9 +589,9 @@ impl Pruner {
...
+        let mut repack_size: BlobTypeMap<u64> = BlobTypeMap::default();
```
Comment (aawsome) [inline src/commands/prune.rs:594]: Can't we remove the type hint here?

File: src/commands/prune.rs [hunk 3]
```diff
@@ -589,9 +589,9 @@ impl Pruner {
         };
 
         self.repack_candidates.sort_unstable_by_key(|rc| rc.0);
-        let mut resize_packs: BlobTypeMap<Vec<_>> = Default::default();
-        let mut do_repack: BlobTypeMap<bool> = Default::default();
-        let mut repack_size: BlobTypeMap<u64> = Default::default();
+        let mut resize_packs: BlobTypeMap<Vec<_>> = BlobTypeMap::default();
```
Comment (aawsome) [inline src/commands/prune.rs:592]: Is there a possibility to simply the type hint now that we are using `BlobTypeMap::default`?

**Review by aawsome — COMMENTED** [review]
Thanks a lot @Dr-Emann for this PR.
Besides minor comments one question: Should we add some of these lints to `src/main.rs`

File: src/commands/prune.rs (hunk 2 above)
Comment (Dr-Emann) [reply src/commands/prune.rs:594]: Unfortunately, only `do_repack` can automatically be inferred as BlobTypeMap<bool>, leaving off the type hints for `resize_packs` and `repack_size` leads to a compile error that rust can't infer the type.

1. Would you prefer leaving off the type hint for do_repack?
2. We could use e.g. `BlobTypeMap::<Vec<_>>::default()`, which do you prefer?

Comment (aawsome) [reply src/commands/prune.rs]: I removed all type hints, using your second option where needed

**Comment by aawsome** [comment]
I updated the `default()` suggestions and reformated `main.rs` to make the format linter happy ;-)

**Review by aawsome — APPROVED** [review]
LGTM - thanks a lot @Dr-Emann, good work!

**Pull request merged**